### ISSUE TITLE
Improve create-test script

### ISSUE THREAD

**Issue by xjamundx** [eslint/espree#291]
I've been having some issues with the `create-test` script.

1. When there is an error message aren't including the `message` property, so they're not passing the tests.
2. The result files include `start` and `end` nodes which aren't allowed.

Here's how I fixed these locally:

```js
 try {
        result = esprima.parse(sourceCode, {
            ecmaVersion: 8,
            ecmaFeatures: {
                experimentalObjectRestSpread: true
            },
            sourceType: 'module', // change to script if needed
            loc: true,
            range: true,
            tokens: true
        });
    } catch (ex) {
        result = {
            message: ex.message,
            column: ex.column,
            index: ex.index,
            lineNumber: ex.lineNumber
        };
    }

    recursivelyRemoveStartAndEnd(result)
    function recursivelyRemoveStartAndEnd(o) {
        if (Array.isArray(o)) {
            o.forEach(recursivelyRemoveStartAndEnd)
        } else if (o && typeof o === 'object') {
            delete o.start
            delete o.end
            Object.keys(o).filter(key => key !== 'loc').forEach((key) => {
                recursivelyRemoveStartAndEnd(o[key])
            })
        }
    }
```

Let me know if I can make this a PR.

**Comment by nzakas** [comment]
I think tests default "script" rather than "module", otherwise :+1:

**Comment by xjamundx** [comment]
Oh yeah I figured that out eventually. Let me update and file a PR :)
